src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,349 @@
+    {
+        private readonly AndroidBrokerHelper _brokerHelper;
+        private readonly ICoreLogger _logger;
+        private Activity _parentActivity;
+        private string _negotiatedBrokerProtocalKey = String.Empty;
+
+        public AndroidContentProviderBroker(CoreUIParent uiParent, ICoreLogger logger)
+        {
+            _parentActivity = uiParent?.Activity;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            AuthenticationContinuationHelper.LastRequestLogger = _logger;
+            _brokerHelper = new AndroidBrokerHelper(Application.Context, logger);
+        }
+
+        public bool IsBrokerInstalledAndInvokable()
+        {
+            return _brokerHelper.IsBrokerInstalledAndInvokable();
+        }
+
+        public async void InitiateBrokerHandshakeAsync()
+        {
+            using (_logger.LogMethodDuration())
+            {
+                try
+                {
+                    Bundle HandshakeBundleResult = await GetHandshakeBundleResultFromBrokerAsync().ConfigureAwait(false);
+
+                    _negotiatedBrokerProtocalKey = _brokerHelper.GetProtocalKeyFromHandshakeResult(HandshakeBundleResult);
+                }
+                catch
+                {
+                    throw;
+                }
+            }
+        }
+
+        private async Task<Bundle> GetHandshakeBundleResultFromBrokerAsync()
+        {
+            var bundle = _brokerHelper.GetHandshakeOperationBundle();
+            var OperationBundleJSON = _brokerHelper.SearializeBundleToJSON(bundle);
+            return await PerformContentResolverOperationAsync(ContentResolverOperation.hello, OperationBundleJSON).ConfigureAwait(false);
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+            _brokerHelper.CheckPowerOptimizationStatus();
+
+            InitiateBrokerHandshakeAsync();
+
+            var brokerRequest = PrepareInteractiveBrokerRequest(authenticationRequestParameters, acquireTokenInteractiveParameters);
+
+            return await PerformAcquireTokenInteractiveAsync(brokerRequest).ConfigureAwait(false);
+        }
+
+        private BrokerRequest PrepareInteractiveBrokerRequest(AuthenticationRequestParameters authenticationRequestParameters,
+                                                              AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+            BrokerRequest brokerRequest = BrokerRequest.FromInteractiveParameters(
+                authenticationRequestParameters, acquireTokenInteractiveParameters);
+
+            // There can only be 1 broker request at a time so keep track of the correlation id
+            AndroidBrokerHelper.CorrelationId = brokerRequest.CorrelationId;
```

Comment:
I'd actually move all the statics to a separate class to simplify this helper a bit.